hazelcast/src/test/java/com/hazelcast/client/impl/protocol/util/ClientMessageEncoderDecoderTest.java

```diff
@@ -0,0 +1,225 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.util;
+
+import com.hazelcast.client.impl.MemberImpl;
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.protocol.codec.MapPutCodec;
+import com.hazelcast.client.impl.protocol.codec.ClientAuthenticationCodec;
+import com.hazelcast.cluster.Member;
+import com.hazelcast.internal.networking.HandlerStatus;
+import com.hazelcast.internal.serialization.impl.HeapData;
+import com.hazelcast.internal.util.counters.SwCounter;
+import com.hazelcast.nio.Address;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import com.hazelcast.version.MemberVersion;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.hazelcast.client.impl.protocol.ClientMessage.FINAL;
+import static com.hazelcast.client.impl.protocol.ClientMessage.UNFRAGMENTED_MESSAGE;
+import static com.hazelcast.internal.networking.HandlerStatus.CLEAN;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ClientMessageEncoderDecoderTest extends HazelcastTestSupport {
+
+
+    @Test
+    public void test() {
+        ClientMessage message = ClientMessage.createForEncode();
+        message.add(new ClientMessage.Frame(new byte[100], UNFRAGMENTED_MESSAGE | FINAL));
+        message.setMessageType((short) MapPutCodec.REQUEST_MESSAGE_TYPE);
+        AtomicReference<ClientMessage> reference = new AtomicReference<>(message);
+
+
+        ClientMessageEncoder encoder = new ClientMessageEncoder();
+        encoder.src(() -> reference.getAndSet(null));
+
+        ByteBuffer buffer = ByteBuffer.allocate(1000);
+        buffer.flip();
+        encoder.dst(buffer);
+
+        HandlerStatus result = encoder.onWrite();
+
+        assertEquals(CLEAN, result);
+
+        AtomicReference<ClientMessage> resultingMessage = new AtomicReference<>();
+        ClientMessageDecoder decoder = new ClientMessageDecoder(null, resultingMessage::set);
+        decoder.setNormalPacketsRead(SwCounter.newSwCounter());
+
+        buffer.position(buffer.limit());
+
+        decoder.src(buffer);
+        decoder.onRead();
+
+        assertEquals(message.getMessageType(), resultingMessage.get().getMessageType());
```

Comment:
Why not verify other fields of the message as well? If the type is the only field to verify, you can better change the test name `testMessageType`?

Reply:
I think we can remove this test completely. It was added when it was not possible to run other tests when the initial prototype is implemented.